app/controllers/shipit/api/stacks_controller.rb

```diff
@@ -40,6 +40,7 @@ def create
       end
 
       params do
+        accepts :environment, String
```

Comment:
Out of curiosity, if we have different shipit.ymls in the repo, and you switch to a new name, does it correctly refresh and pull down the "new" one (`shipit.newenvname.yml`)?

If not, it may be worth enqueueing a refresh when this field is updated.